docs/use/api-importation.rst

```diff
@@ -0,0 +1,59 @@
+.. _starting-a-tournament:
+
+=====================
+Import Data Using API
+=====================
+
+It is possible to import data from third-party registration system using POST method. The APIs are as follows
+
+Adding Institutions
+========================
+/<tournament code>/admin/participants/addspeaker/
+
+- username
+    Username of a staff account
+- password:
+    Password of the staff account
+- inst_name:
+    Institution name
+- inst_code:
+     Institution code
+
+Adding Teams
+==============================
+/<tournament code>/admin/participants/addspeaker/
+
+- username
+    Username of a staff account
+- password:
+    Password of the staff account
+- code_name:
+    Team code name for identification, it must be unique
+- reference
+    Reference of the team
+- institution
+    The name of institution
+- type (Optional)
```

Comment:
> Hey! This is great! Thanks for this work. I've left a few comments, of which should be generalizable to the extent of the whole patch.
> 
> I must say this is not quite what I expected with #1301; where we would be using django REST framework (like what we already do with serializers) to get these POST views. What I had in mind were three JSON endpoints:
> 
> * adjudicators:
>   
>   * name
>   * email (optional)
>   * institution (optional; institution code)
>   * base score (default 0.0)
>   * gender (optional)
>   * pronoun (optional)
> * institutions:
>   
>   * name
>   * code
> * teams:
>   
>   * reference (called name; short_reference created from that)
>   * uses_institution_prefix (default true)
>   * code (optional)
>   * institution (code)
>   * speakers (as list)
>     
>     * name
>     * email
>     * gender
> 
> The big difference would be merging the teams/speaker endpoints.
> 
> Comments on this conception of the import API?

Using JSON instead of Form Post does make more sense as it allows multiple data submissions in one POST request. Speaking of which, I find it better to use one endpoint for data submission instead of three as it allows the client to submit all relevant information regarding the registration of one institution in one POST request.

Here is a sample JSON data 

`{"institutions":[
	{ "name": "PK UNIVERSITY", "code":"PKU" }
	],
"teams":[
	{ "name": "PKU A", "uses_institution_prefix":"TRUE","code":"PKU A CODE","institution":"PKU",
	"speakers":[
		{"name":"Jason Wang", "email":"123@126.com","gender":"M"}
		]
	}
	]}`

Reply:
Great! One thing: If using the institution prefix, the reference would just be "A", not "PKU A". Looks awesome!